### PR TITLE
chore: Update interdependent dependencies and revise command-line interface

### PR DESCRIPTION
**Pull request by gwhitney**
Updates click, Flask and several of its extensions, itsdangerous, Jinja2,
  MarkupSafe, SQLAlchemy, and Werkzeug.

  Removes the obsolete Flask-Script extension, as that has been superseded
  by the built-in Flask command-line interface `flask <COMMAND> ...`

  NOTE: interacting with the server must henceforth occur just
  with the new `flask <COMMAND> ...` interface.
  Updates the docs to remove all references to manage.py.
  Reimplements `python manage.py test` as `flask test`.
  Makes progress on https://github.com/numberscope/backscope/issues/43.
  Supersedes https://github.com/numberscope/backscope/pull/63.
  Supersedes https://github.com/numberscope/backscope/pull/90.

### CODE REVIEW

**Comment by gwhitney** [comment]
Even though the primary point here is "just" to update dependencies, and the test did still run correctly for me, there have been pretty significant changes under the hood, to the extent that the old "manage.py" setup for interacting with the server has been eliminated. Hence this definitely needs manual review and testing by another party before it should be merged. And note (while testing and after merging this) we all need to shift to use `flask <COMMAND> ...` to manipulate the backscope (the Flask-Script extension that supported manage.py was retired between our old version of Flask and the current one). Thanks!

File: doc/tests.md [hunk 1]
```diff
@@ -45,9 +45,9 @@ For guidance, consult the basic instructions on how to [set up your environment]
 1. Go into the top-level directory of the Backscope repository.
 2. Activate the Backscope virtual environment.
    + If you're using [`venv`](https://docs.python.org/3/library/venv.html) to manage virtual environments, and you've put Backscope's virtual environment in a directory called `.venv`, use the command `source .venv/bin/activate` to activate.
-3. Call `python manage.py test`.
+3. Execut the command `flask test`.
```
Comment (katestange) [inline doc/tests.md:48]: spelling

**Review by katestange — CHANGES_REQUESTED** [review]
I installed the dependencies, ran `flask run` and `flask test` as well as the various `flask db` commands, and all appeared to be working (although I didn't blow away the database, so I was just checking they appear to call the right things, even though they told me I already had a migration etc.).  There's one typo I pointed out.  Do I need to anything else?

**Comment by gwhitney** [comment]
Fixed the typo. If you have done the pip install and can run everything, and the backscope works on your machine when you run it, and you've reviewed the code changes (as seems clear by catching the typo), then no, I can't think of anything else you need to do. Thanks!

**Review by katestange — APPROVED** [review]

**Pull request merged**